pkg/vault/test/foundry/RecoveryMode.t.sol

```diff
@@ -49,6 +51,39 @@ contract RecoveryModeTest is BaseVaultTest {
         router.removeLiquidityRecovery(pool, amountToRemove);
     }
 
+    function testRecoveryModeWithRoundtripFee() public {
+        // Add initial liquidity.
+        uint256[] memory amountsIn = [uint256(defaultAmount), uint256(defaultAmount)].toMemoryArray();
+
+        vm.prank(alice);
+        (, uint256 bptAmountOut, ) = router.addLiquidityCustom(pool, amountsIn, bptAmount, false, bytes(""));
+
+        // Put pool in recovery mode.
+        vault.manualEnableRecoveryMode(pool);
+        // Set the fee, and flag to trigger collection of it.
+        vault.manualSetStaticSwapFeePercentage(pool, BASE_MAX_SWAP_FEE);
+        vault.manualSetAddLiquidityCalledFlag(pool, true);
+
+        uint256 initialSupply = IERC20(pool).totalSupply();
+        uint256 amountToRemove = bptAmountOut / 2;
+        uint256 amountOutWithoutFee = defaultAmount / 2;
+        uint256 feeAmount = amountOutWithoutFee.mulDown(BASE_MAX_SWAP_FEE);
+        uint256 amountOutAfterFee = amountOutWithoutFee - feeAmount;
+
+        vm.expectEmit();
+        emit IVaultEvents.PoolBalanceChanged(
+            pool,
+            alice,
+            initialSupply - amountToRemove, // totalSupply after the operation
+            [-int256(amountOutAfterFee), -int256(amountOutAfterFee)].toMemoryArray(),
+            [feeAmount, feeAmount].toMemoryArray()
+        );
+
+        // Do a recovery withdrawal.
+        vm.prank(alice);
+        router.removeLiquidityRecovery(pool, amountToRemove);
```

Comment:
Can we please also check balances? I know it's too much, but we're at a very late stage now.